src/System.Private.ServiceModel/src/System/ServiceModel/Channels/RTSocketConnection.cs

```diff
@@ -497,6 +505,10 @@ protected override void TraceSocketReadStop(int bytesRead, bool async)
                     throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(exceptionToThrow);
                 }
             }
+            catch (OperationCanceledException)
+            {
+                throw;
```

Comment:
Same questions about why this is needed and whether we need to AbortWrite().  Not blocking, just questions.